lib/modules/platform/github/index.ts

```diff
@@ -731,33 +731,50 @@ export async function getBranchForceRebase(
   branchName: string,
 ): Promise<boolean> {
   config.branchForceRebase ??= {};
-  if (config.branchForceRebase[branchName] === undefined) {
-    try {
-      config.branchForceRebase[branchName] = false;
-      const branchProtection = await getBranchProtection(branchName);
-      logger.debug(`Found branch protection for branch ${branchName}`);
-      if (branchProtection?.required_status_checks?.strict) {
-        logger.debug(
-          `Branch protection: PRs must be up-to-date before merging for ${branchName}`,
-        );
-        config.branchForceRebase[branchName] = true;
-      }
-    } catch (err) {
-      if (err.statusCode === 404) {
-        logger.debug(`No branch protection found for ${branchName}`);
-      } else if (
-        err.message === PLATFORM_INTEGRATION_UNAUTHORIZED ||
-        err.statusCode === 403
-      ) {
-        logger.once.debug(
-          'Branch protection: Do not have permissions to detect branch protection',
-        );
-      } else {
-        throw err;
-      }
+
+  const cachedResult = config.branchForceRebase[branchName];
+  if (cachedResult !== undefined) {
+    return cachedResult;
+  }
+
+  // Initialize to false before checking branch protection
+  config.branchForceRebase[branchName] = false;
+
+  try {
+    const branchProtection = await getBranchProtection(branchName);
+    logger.debug(`Found branch protection for branch ${branchName}`);
+
+    const strictStatusChecks = branchProtection?.required_status_checks?.strict;
+    if (strictStatusChecks) {
+      logger.debug(
+        `Branch protection: PRs must be up-to-date before merging for ${branchName}`,
+      );
+      config.branchForceRebase[branchName] = true;
     }
+  } catch (err) {
+    handleBranchProtectionError(err, branchName);
   }
-  return !!config.branchForceRebase[branchName];
+
+  return config.branchForceRebase[branchName];
+}
+
+function handleBranchProtectionError(err: any, branchName: string): void {
+  if (err.statusCode === 404) {
+    logger.debug(`No branch protection found for ${branchName}`);
+    return;
+  }
+
+  const isUnauthorized =
+    err.message === PLATFORM_INTEGRATION_UNAUTHORIZED || err.statusCode === 403;
+
+  if (isUnauthorized) {
+    logger.once.debug(
```

Comment:
Is this intentional?

Reply:
Yes, this is copied from https://github.com/renovatebot/renovate/pull/37922/files#diff-8261f11faec10d2c67d98fb86dfc74f0d6bd1999aa79eb3464c8898988854449L752